src/main/java/com/aws/greengrass/secretmanager/Result.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.secretmanager;
+
+public class Result<T> {
+    private boolean set;
+    private T value;
+
+    public Result() {
+    }
+
+    public Result(T t) {
+        this.value = t;
+        this.set = true;
+    }
+
+    /**
+     * Set the value.
+     *
+     * @param t new value
+     * @return this
+     */
+    public synchronized Result<T> set(T t) {
+        this.value = t;
+        this.set = true;
+        this.notifyAll();
+        return this;
+    }
+
+    public boolean isSet() {
```

Comment:
shouldn't `isSet` and `getValue` also synchronized?

Reply:
No, the synchronization isn't for reading/writing fields. It is used for `notifyAll` and `wait`, when using notify and wait, the object must be sychronized

Reply:
I can add it, but it isn't required for anything.